system/jlib/jthread.cpp

```diff
@@ -2005,16 +2005,51 @@ protected: friend class PipeWriterThread;
         return argv;
     }
 
+    bool isPlatformProcess(HANDLE pid)
+    {
+        // MORE - may need to cope with other installation locations - perhaps check that the installation location matches the current exe's location? Is there some other way to know the install location?
+        VStringBuffer procPath("/proc/%u/exe", pid);
+        char path[PATH_MAX + 1];
+        ssize_t len = readlink(procPath, path, PATH_MAX);
+        if (len != -1)
+        {
+            path[len] = 0;
+            return startsWith(path, "/opt/HPCCSystems/bin/");
```

Comment:
Yea, this would not work if /opt was a soft link to another dir as we saw in a UK install recently.
Perhaps a regex, includes("HPCCSystems") ?
Maybe check the environ of the pid or parent for something unique (or set something unique to know what to check for).
Is this an added safety check for not writing the secret data out the new pipe ?